cheroot/server.py

```diff
@@ -1220,15 +1219,13 @@ def communicate(self):
                 if self.server.stats['Enabled']:
                     self.requests_seen += 1
                 if not req.ready:
-                    # Something went wrong in the parsing (and the server has
-                    # probably already made a simple_response). Return and
-                    # let the conn close.
-                    return
+                    # Allow conn to stay open to support non-blocking sockets
+                    return True
 
                 request_seen = True
                 req.respond()
                 if req.close_connection:
-                    return
+                    return False
```

Comment:
The first thing I notice is that these changes seem to be altering the return value of parse_request, to return True or False. I don't see where the meaning of this return value is documented. Can you document it? More importantly, why is it that nothing is using this return value?